examples/platforms/efr32/uart.c

```diff
@@ -227,14 +277,16 @@ otError otPlatUartDisable(void)
 otError otPlatUartSend(const uint8_t *aBuf, uint16_t aBufLength)
 {
     otError error = OT_ERROR_NONE;
+    Ecode_t status;
 
     otEXPECT_ACTION(sTransmitBuffer == NULL, error = OT_ERROR_BUSY);
 
     sTransmitBuffer = aBuf;
     sTransmitLength = aBufLength;
 
-    UARTDRV_Transmit(sUartHandle, (uint8_t *)sTransmitBuffer, sTransmitLength, transmitDone);
-
+    status = UARTDRV_Transmit(sUartHandle, (uint8_t *)sTransmitBuffer, sTransmitLength, transmitDone);
+    assert(ECODE_OK == status);
```

Comment:
I had look at the conditions in which a failure could be returned from these UARTDRV functions, and they seem to centre around bad parameters passed, or not enough buffers defined, rather than sporadically occurring errors caused by something external during runtime - so I figured an assert would be most useful for us to catch issues caused by the programmer like this rather than silently returning an error code.

On the flip side, I also added returning an error code after the assert to ensure the error is still handled if OpenThread is compiled without asserts.

What are your thoughts?

Reply:
Seems sensible approach.  As you say the errors are really for validation of parameters and an assert will catch these.